codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientClass.java

```diff
@@ -341,6 +347,26 @@ private MethodSpec traditionalMethod(OperationModel opModel) {
         return method.build();
     }
 
+    public static Optional<CodeBlock> addRequestModifierCode(OperationModel opModel, IntermediateModel model) {
+        Map<String, RequestCustomizer> requestCustomizerMap = model.getCustomizationConfig().getRequestCustomizerMap();
+        if (!CollectionUtils.isNullOrEmpty(requestCustomizerMap)) {
```

Comment:
In this particular function, we have two cases where we return Optional.empty():

1. CollectionUtils.isNullOrEmpty(requestCustomizerMap)
2.  customRequestTransformerMap.get(opModel.getOperationName()) == null

Following the reverse if logic will result in duplicate return Optional.empty();. The code will look like this:
```java
    public static Optional<CodeBlock> addRequestModifierCode(OperationModel opModel, IntermediateModel model) {
        Map<String, CustomRequestTransformer> customRequestTransformerMap =
            model.getCustomizationConfig().getCustomRequestTransformerMap();

        if (CollectionUtils.isNullOrEmpty(customRequestTransformerMap)) {
            return Optional.empty();
        }
        CustomRequestTransformer customRequestTransformer = customRequestTransformerMap.get(opModel.getOperationName());
        if (customRequestTransformer == null) {
            return Optional.empty();
        }
        CodeBlock.Builder builder = CodeBlock.builder();
        ClassName instanceType = classNameFromFqcn(customRequestTransformer.getClassName());
        builder.addStatement("$L = $T.$N($L)",
                             opModel.getInput().getVariableName(),
                             instanceType,
                             customRequestTransformer.getMethodName(),
                             opModel.getInput().getVariableName());
        return Optional.of(builder.build());
    }
```

Since there are two cases where we need to return Optional.empty(), avoiding code duplication is preferable for better readability.